packages/web/components/DigestEmailConfigurationFormField/DigestEmailConfigurationFormField.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react'
+import {
+  useAddUserInterestMutation,
+  DigestEmailConfiguration as DigestEmailConfigurationType,
+} from '@/generated/graphql'
+import Button, { ButtonVariant } from '@/components/Button'
+import { useTranslation } from '@/config/i18n'
+import Select from '../Select'
+
+type Props = {
+  digestEmailConfig: DigestEmailConfigurationType
+  refetch: () => void
+}
+
+const DigestEmailConfigurationFormField: React.FC<Props> = ({ digestEmailConfig, refetch }) => {
+  const { t } = useTranslation('settings')
+
+  // The filter operation here is just to remove the "WEEKLY" option
+  // until we build support for it
+  const options = (Object.keys(DigestEmailConfigurationType) as (keyof typeof DigestEmailConfigurationType)[]).map(
+    (key, index) => ({
+      value: index,
+      displayName: DigestEmailConfigurationType[key],
+    }),
+  ).filter((option) => option.displayName !== DigestEmailConfigurationType.Weekly);
+
+  // TODO: Figure out why this is -1 even though loggig everything before the OR returns 0
+  const currentConfig = options.find((option) => option.displayName === digestEmailConfig)?.value || -1
+  const [selectedOptionId, setSelectedOptionId] = useState<number>(currentConfig)
+  console.log(digestEmailConfig)
+  const [
+    addUserInterest,
+    { loading: addingUserInterest },
+  ] = useAddUserInterestMutation()
```

Comment:
We probably want to change the pattern from what's being copied here so that `NotificationSettingsForm` owns the state and is responsible for patching the settings record. Not sure if you want "commit on change" semantics or for there to be an explicit save CTA here, but either way having a mutation and logic-heavy component per-setting seems like something we'd want to avoid.

Reply:
Good call!